toucan_connectors/google_big_query/google_big_query_connector.py

```diff
@@ -53,24 +67,39 @@ class GoogleBigQueryConnector(ToucanConnector):
         '<a href="https://developers.google.com/identity/protocols/googlescopes" target="_blank" >documentation</a>',
     )
 
-    def _retrieve_data(self, data_source: GoogleBigQueryDataSource) -> pd.DataFrame:
-        """
-        Uses Pandas read_gbq method to extract data from Big Query into a dataframe
-        See: http://pandas.pydata.org/pandas-docs/stable/generated/pandas.io.gbq.read_gbq.html
-        Note:
-            The parameter reauth is set to True to force Google BigQuery to reauthenticate the user
-            for each query. This is necessary when extracting multiple data to avoid the error:
-            [Errno 54] Connection reset by peer
-        """
+    @staticmethod
+    def _get_google_credentials(credentials, scopes):
+        credentials = get_google_oauth2_credentials(credentials).with_scopes(scopes)
+        return credentials
+
+    @staticmethod
+    def _connect(credentials) -> bigquery.Client:
+        start = timer()
+        client = bigquery.Client(credentials=credentials)
+        end = timer()
+        logging.getLogger(__name__).info(
+            f'[benchmark][google_big_query] - connect {end - start} seconds',
+            extra={
+                'benchmark': {
+                    'operation': 'connect',
+                    'execution_time': end - start,
+                    'connector': 'google_big_query',
+                }
+            },
+        )
+        return client
+
+    @staticmethod
+    def _execute_query(client: bigquery.Client, query: str, parameters: List) -> pandas.DataFrame:
```

Comment:
it's a security check but i was thinking it will be great to have a `try - catch` here, so we can check either or not the query was well perform... because we never knows when communicating with external systems...

maybe this is overdoing too... but am just suggesting...